lib/jekyll/collection.rb

```diff
@@ -100,15 +100,15 @@ def filtered_entries
     # Returns a String containing the directory name where the collection
     #   is stored on the filesystem.
     def relative_directory
-      @relative_directory ||= "_#{label}"
+      @relative_directory ||= Pathname.new(directory).relative_path_from(Pathname.new(site.source)).to_s
```

Comment:
This isn't a breaking change because the functionality is now in the `directory` method below.